pkg/builder/csolution/builder.go

```diff
@@ -206,61 +207,101 @@ func (b CSolutionBuilder) getCprjsBuilders(selectedContexts []string) (cprjBuild
 
 		idxFile, err := b.getIdxFilePath()
 		if err != nil {
-			return cprjBuilders, err
+			return projBuilders, err
 		}
 
-		cprjFile, err := b.getCprjFilePath(idxFile, context)
-		if err != nil {
-			log.Error("error getting cprj file: " + err.Error())
-			return cprjBuilders, err
+		buildOptions := b.Options
+		// Set XML schema check to false, when input is yml
+		if b.Options.Schema {
+			buildOptions.Schema = false
 		}
 
-		// get cprj builder
-		cprjBuilder := cproject.CprjBuilder{
-			BuilderParams: builder.BuilderParams{
-				Runner:         b.Runner,
-				Options:        b.Options,
-				InputFile:      cprjFile,
-				InstallConfigs: b.InstallConfigs,
-			},
-		}
+		var projBuilder builder.IBuilderInterface
+		if b.Options.UseCbuild2CMake {
+			buildOptions.Contexts = []string{context}
+			// get cprj builder
+			projBuilder = cbuildidx.CbuildIdxBuilder{
+				BuilderParams: builder.BuilderParams{
+					Runner:         b.Runner,
+					Options:        buildOptions,
+					InputFile:      idxFile,
+					InstallConfigs: b.InstallConfigs,
+				},
+			}
```

Comment:
It seems a `CbuildIdxBuilder` is being used for every context, in parallel to what is done with CPRJs, but when using `cbuild2cmake` it should be simpler.
The `cbuild2cmake` requires a single call for the whole solution, as well as the CMake configuration step. Even context specific calls `cmake --build tmp --target <context>` should be grouped if possible `cmake --build tmp --target <context1> <context2> <...>`

Reply:
I see the feasibility to simplify `cbuild2cmake` usage with a single call for configuration and execution becomes more feasible. However, when it comes to grouping targets using `cmake --build tmp --target <context1> --target <context2> --target <...>`, the potential downside lies in difficulty identifying which context failed to build if multiple targets encounter failures. This is because `cmake `generates a unified log for all target builds when grouping them.

To address this issue and ensure separate logs for specific contexts, I propose:

- Utilizing a single command for the entire solution.
- Employing a single command for configuration.
- Implementing single or multiple context-specific build commands `cmake --build tmp --target <context>`.

This approach should facilitate clearer identification of build failures within specific contexts.

Reply:
The CMakeLists generated by `cbuild2cmake` allow CMake to automatically build dependent targets and decide the correct build order. Dependent targets would anyway appear in an "unexpected" section of the log in case of multiple CMake build calls.
Talking with Reinhard and Joachim we believe a single CMake build call also for multiple contexts should be implemented initially, afterwards we can see whether the log needs refinement.

Reply:
[Addressed review commments](https://github.com/Open-CMSIS-Pack/cbuild/pull/167/commits/509e86590c99aa321022ac80d1ba92dd21073935)